### PR TITLE
golang: update to version 1.15.11 to address CVE-2021-27918

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

golang: update to version 1.15.11 to address CVE-2021-27918

###### Change Log  
golang spec file and all specs which build requires golang

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-27918

### CODE REVIEW

File: SPECS/golang/golang-1.15.spec [hunk 1]
```diff
@@ -115,6 +115,9 @@ fi
 %{_bindir}/*
 
 %changelog
+* Mon Apr 26 2021 Nicolas Guibourge <nicolasg@microsoft.com> - 1.15.11-1
+- Updated to version 1.15.1 to fix CVE-2021-27918
```
Comment (anphel31) [inline SPECS/golang/golang-1.15.spec:119]: typo - should be 1.15.11

File: SPECS/runc/runc.spec [hunk 2]
```diff
@@ -84,6 +84,8 @@ make install BINDIR=%{buildroot}%{_sbindir}
 %{_sbindir}/runc
 
 %changelog
+* Mon Apr 26 2021 Nicolas Guibourge <nicolasg@microsoft.com> 1.0.0.rc8-8
```
Comment (anphel31) [inline SPECS/runc/runc.spec:87]: nit: indentation

File: SPECS/telegraf/telegraf.spec [hunk 3]
```diff
@@ -80,6 +80,8 @@ fi
 %dir %{_sysconfdir}/%{name}/telegraf.d
 
 %changelog
+* Mon Apr 26 2021 Nicolas Guibourge <nicolasg@microsoft.com> 1.14.5-6
```
Comment (anphel31) [inline SPECS/telegraf/telegraf.spec:83]: nit: indentation

**Review by anphel31 — APPROVED** [review]

File: SPECS/golang/golang-1.15.spec (hunk 1 above)
Comment (nicogbg) [reply SPECS/golang/golang-1.15.spec:119]: ooops thanks

File: SPECS/runc/runc.spec (hunk 2 above)
Comment (nicogbg) [reply SPECS/runc/runc.spec:87]: fixed

File: SPECS/telegraf/telegraf.spec (hunk 3 above)
Comment (nicogbg) [reply SPECS/telegraf/telegraf.spec:83]: fixed

File: SPECS/etcd/etcd-3.4.3.spec [hunk 4]
```diff
@@ -74,6 +74,8 @@ rm -rf %{buildroot}/*
 %config(noreplace) %{_sysconfdir}/etcd/etcd-default-conf.yml
 
 %changelog
+*   Mon Apr 26 2021 Nicolas Guibourge <nicolasg@microsoft.com> 3.4.3-2
+-   Increment release to force republishing using golang 1.15.11.
```
Comment (PawelWMS) [inline SPECS/etcd/etcd-3.4.3.spec:78]: Why are we incrementing these everywhere? Are these packages linking to some `golang` libs and the libs version changed?

Comment (nicogbg) [reply SPECS/etcd/etcd-3.4.3.spec:78]: these packages have a build require dependency on golang...

Comment (nicogbg) [reply SPECS/etcd/etcd-3.4.3.spec:78]: if I recall correctly etcd is written in go

Comment (PawelWMS) [reply SPECS/etcd/etcd-3.4.3.spec:78]: But it doesn't seem to have any run-time dependency on its libraries (if `golang` even provides any). It's not a problem to rebuild these other packages, it's just surprising. We didn't do that when we updated other packages as far. For instance, we do update `glib` or `gcc` every now and then, but don't update all specs relying on them. Do you think there's a reason we should be doing that and we're missing this from our process?

**Review by PawelWMS — APPROVED** [review]

File: SPECS/etcd/etcd-3.4.3.spec (hunk 4 above)
Comment (PawelWMS) [reply SPECS/etcd/etcd-3.4.3.spec:78]: Thanks for the clarification, Nicolas! I thought about it more and see your point.

**Pull request merged**